juspay/hyperswitch
# feat(core): add SCA exemption field

## Type of Change



## Description

- To skip authorization, SCA exemption field is added to the transaction to flag it from authentication for frictionless flow for the user. 
- The field is passed during the request and accept 2 values: `low_value` and, `transaction_risk_analysis`

### Additional Changes

Add a new field in payment request
```
{
    "psd2_sca_exemption_type": "low_value",
}
```
Add a new field to the payment intent table: `psd2_sca_exemption_type`




## Motivation and Context



## How did you test it?

Tested through postman:
- Create a MCA (Cybersource):
- Create a /payment call
```
{
    "amount": 499,
    "currency": "USD",
    // "confirm": true,
    "capture_method": "automatic",
    "customer_id": "test_rec7",
    "email": "guest@example.com",
    "request_external_three_ds_authentication": true,
    "customer_acceptance": {
        "acceptance_type": "online"
    },
    "payment_method": "card",
    "payment_method_type": "credit",
    "payment_method_data": {
        "card": {
            "card_number": "4242424242424242",
            "card_exp_month": "03",
            "card_exp_year": "2030",
            "card_holder_name": "Test",
            "card_cvc": "737"
        }
    },
    "billing": {
        "address": {
            "city": "test",
            "country": "US",
            "line1": "here is some \n there is some \n none is some? \n ",
            "line2": "there",
            "line3": "anywhere",
            "zip": "560095",
            "state": "Washington",
            "first_name": "One",
            "last_name": "Last"
        },
        "phone": {
            "number": "1234567890",
            "country_code": "+1"
        },
        "email": "guest@example.com"
    },
     "psd2_sca_exemption_type": "low_value",
    "authentication_type": "no_three_ds"
}
```
- `psd2_sca_exemption_type` field should be present in the payment_intent table against the generated payment_id

## Checklist
